ili/utils/ndes_pt.py

```diff
@@ -262,41 +268,52 @@ def load_nde_lampe(
             Defaults to True.
         theta_normalize (bool, optional): whether to z-normalize theta.
             Defaults to True.
+        engine (str, optional): dummy argument to match sbi interface.
+            Must be set to 'NPE' or will be overwritten.
         **model_args: additional arguments to pass to the model.
     """
+    if 'NPE' not in engine:
+        logging.warning(f'Engine {engine} not supported in lampe backend. '
+                        'Continuing as if to engine=NPE.')
+
     if model == 'mdn':  # for mixture density networks
         if not (set(model_args.keys()) <= {'hidden_features', 'num_components'}):
             raise ValueError(f"Model {model} arguments mispecified.")
+        model_defaults = dict(hidden_features=16, num_components=3)
+        model_args = {**model_defaults, **model_args}
```

Comment:
Given that the if statement above checks that the given model_args contain both hidden_features and num_components, I don't think these model_defaults will do anything, as they will be immediately overwritten in this line. So if you want to use the defaults in the case when the user doesn't specify 'hidden_features' or 'num_components', then the if statement should be removed. 

Also, the ValueError from the if statement is never raised in the unit tests, so (if we keep the if statement), please add a check for this.

Reply:
This doesn't check that model_args must contain both hidden_features and num_components, it simply checks whether model_args contains anything extra besides hidden_features and num_components (if its a subset).

I've updated the error to make this clearer and added a test for it